enhancements/hypershift/topology-aware-scheduling/topology-aware-scheduling.md

```diff
@@ -0,0 +1,247 @@
+on HCP is similar to the deployment and usage on OCP. 
+No major changes are necessary on that front.
+
+### API Extensions
+
+Extend the [NUMAResourceOperator](https://github.com/openshift-kni/numaresources-operator/blob/0beeb62c7c0182ba5a2963ccd658bd0f6af44ebd/api/numaresourcesoperator/v1/numaresourcesoperator_types.go#L164) CR to contain NodeSelector filed.
+`spec.NodeGroup[].NodeSelector`
+User can choose the group of node by the NodeSelector and to have a unified API for OCP and HCP.
+
+For example:
+
+  OpenShift:
+
+  `spec.NodeGroup[].NodeSelector = "node-role.kubernetes.io/worker="`
+  
+  HyperShift:
+
+  `spec.NodeGroup[].NodeSelector = "hypershift.openshift.io/nodePool=hostedcluster02"`
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This proposal aims for Hypershift / Hosted Control Planes alone. 
+
+#### Standalone Clusters
+
+* N/A
+
+#### Single-node Deployments or MicroShift
+
+* N/A
+
+### Implementation Details/Notes/Constraints
+
+#### Expose KubeletConfig to Hosted-Clusters
+
+At first, when a user associates a KubeletConfig's ConfigMap to the NodePool,
+the NodePool controller mirrors the ConfigMap to the control-plane namespace.
+Similar to: https://github.com/openshift/hypershift/pull/4150
+The KubeletConfig's ConfigMap NamespacedName convention, created by the NodePool controller is:
+`<control-plane-namespace>/<name+node-pool>`
+* `control-plane-namespace` = name of the control-plane namespace
+* `name` = name of the ConfigMap as provided by the user in the hosted-cluster namespace.
+* `node-pool` = name of the NodePool on which the ConfigMap was referenced.
+
+To make sure the final name is valid,
+it's provided to the https://github.com/openshift/hypershift/blob/ffa435f71e1a037174e29d31a2b80d569c940d57/support/util/route.go#L39 so that 
+```
+finalName := ShortenName(name, NodePoolName)
+``` 
+
+Then, HCCO picks it up from the control-plane namespace and copies it to the hosted-cluster.
+
+The KubeletConfig's ConfigMap NamespacedName convention, created by HCCO is:
+`openshift-config-managed/<name>`
+* `openshift-config-managed` = name of the namespace where all the configs managed by OCP reside.
+* `name` = name of the ConfigMap as provided by the NodePool controller in the control-plane namespace.
+
+In addition, KubeletConfig ConfigMap contains the following labels/annotations:
+
+Labels:
+* `hypershift.openshift.io/kubeletconfig-config` : `""`
+* `hypershift.openshift.io/nodePool` : `NodePool` API name where the KubeletConfig has referenced
+
+Annotations:
+* `hypershift.openshift.io/nodePool` : `NodePool` API name where the KubeletConfig has referenced
+
+When the KubeletConfig ConfigMap is deleted by the user,
+the hosted-cluster gets scraped and recreated again (which obviously deletes the ConfigMap as well).
+
+Then, the NodePool controller deletes the ConfigMap from the control-plane namespace
+so the HCCO will not try to recreate it when its already gone.
```

Comment:
This is not clear either. How will the HCCO know that it needs to delete a configmap from the openshift-config-managed namespace in the hosted cluster?

Reply:
It doesn't need to. When there's a change in KubeletConfig it will cause a recreation of the cluster. (according to my understanding but maybe I'm wrong here)
A recreation of the cluster will make the remove the ConfigMap from the cluster.

Reply:
Yes, as mentioned above, the cluster is not recreated